Unexpected error occurred while generating suggestions for input: /undo

### Server Implementation

Paper

### Server Version

1.17.1

### Describe the bug

Upon tab completing //undo, the following message gets sent in console

```
[16:35:52 ERROR]: Unexpected error occurred while generating suggestions for input: /undo 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at com.fastasyncworldedit.core.extension.platform.binding.Bindings.invoke(Bindings.java:135) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.fastasyncworldedit.core.extension.platform.binding.Bindings$1.convert(Bindings.java:104) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at org.enginehub.piston.impl.CommandParser.lambda$isAcceptedByTypeParsers$7(CommandParser.java:497) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:?]
        at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812) ~[?:?]
        at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
        at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502) ~[?:?]
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488) ~[?:?]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
        at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[?:?]
        at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[?:?]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632) ~[?:?]
        at org.enginehub.piston.impl.CommandParser.isAcceptedByTypeParsers(CommandParser.java:492) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at org.enginehub.piston.impl.CommandParser.parseRegularArgument(CommandParser.java:448) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at org.enginehub.piston.impl.CommandParser.parse(CommandParser.java:340) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at org.enginehub.piston.impl.CommandManagerImpl.parse(CommandManagerImpl.java:240) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at org.enginehub.piston.impl.CommandManagerImpl.getSuggestions(CommandManagerImpl.java:190) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.extension.platform.PlatformCommandManager.handleCommandSuggestion(PlatformCommandManager.java:912) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.util.eventbus.MethodHandleEventHandler.dispatch(MethodHandleEventHandler.java:51) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.util.eventbus.EventHandler.handleEvent(EventHandler.java:73) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.util.eventbus.EventBus.dispatch(EventBus.java:193) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.util.eventbus.EventBus.post(EventBus.java:181) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.bukkit.WorldEditPlugin$AsyncTabCompleteListener.onAsyncTabComplete(WorldEditPlugin.java:655) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:37) ~[patched_1.17.1.jar:git-Paper-353]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[patched_1.17.1.jar:git-Paper-353]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.17.1.jar:git-Paper-353]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:628) ~[patched_1.17.1.jar:git-Paper-353]
        at org.bukkit.event.Event.callEvent(Event.java:45) ~[patched_1.17.1.jar:git-Paper-353]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleCustomCommandSuggestions(ServerGamePacketListenerImpl.java:815) ~[app:?]
        at net.minecraft.network.protocol.game.ServerboundCommandSuggestionPacket.handle(ServerboundCommandSuggestionPacket.java:28) ~[app:?]
        at net.minecraft.network.protocol.game.ServerboundCommandSuggestionPacket.handle(ServerboundCommandSuggestionPacket.java:6) ~[app:?]
        at net.minecraft.network.Connection.genericsFtw(Connection.java:303) ~[app:?]
        at net.minecraft.network.Connection.channelRead0(Connection.java:289) ~[app:?]
        at net.minecraft.network.Connection.channelRead0(Connection.java:54) ~[app:?]
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.handler.flush.FlushConsolidationHandler.channelRead(FlushConsolidationHandler.java:152) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:795) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:480) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[patched_1.17.1.jar:git-Paper-353]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[patched_1.17.1.jar:git-Paper-353]
        at java.lang.Thread.run(Thread.java:831) ~[?:?]
Caused by: java.lang.reflect.InvocationTargetException
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
        at com.fastasyncworldedit.core.extension.platform.binding.Bindings.invoke(Bindings.java:132) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        ... 71 more
Caused by: com.sk89q.worldedit.internal.expression.ParserException: Parser error at 1: mismatched input '<EOF>' expecting {'+', '-', '~', '++', '--', '(', '{', ';', '!', 'if', 'while', 'do', 'for', 'break', 'continue', 'return', 'switch', NUMBER, ID}
        at com.sk89q.worldedit.internal.expression.ParserErrorListener.syntaxError(ParserErrorListener.java:37) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.antlr4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.antlr4.runtime.Parser.notifyErrorListeners(Parser.java:544) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.antlr4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.antlr4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.antlr.ExpressionParser.statements(ExpressionParser.java:229) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.antlr.ExpressionParser.allStatements(ExpressionParser.java:162) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.internal.expression.Expression.<init>(Expression.java:110) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.sk89q.worldedit.internal.expression.Expression.compile(Expression.java:79) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at com.fastasyncworldedit.core.extension.platform.binding.ConsumeBindings.limitInteger(ConsumeBindings.java:66) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
        at com.fastasyncworldedit.core.extension.platform.binding.Bindings.invoke(Bindings.java:132) ~[FastAsyncWorldEdit-Bukkit-1.17-380.jar:?]
        ... 71 more
```

Potentially related to #1375

### To Reproduce

1. Tab complete //undo

### Expected behaviour

No error

### Screenshots / Videos

_No response_

### Error log (if applicable)

_No response_

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/b6dff701f0dc4542b60540f0bf66a322

### Fawe Version

1.17-380;ee0d1b5

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit-1.17/ and the issue still persists.

### Anything else?

Paper 1.17.1, build 353

--- a/worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/platform/binding/Bindings.java
+++ b/worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/platform/binding/Bindings.java
@@ -101,11 +101,15 @@ public class Bindings {
 
                 @Override
                 public ConversionResult<Object> convert(String s, InjectedValueAccess access) {
-                    Object o = invoke(s, argsFunc, access, method);
-                    if (o == null) {
-                        return FailedConversion.from(new NullPointerException());
+                    try {
+                        Object o = invoke(s, argsFunc, access, method);
+                        if (o == null) {
+                            return FailedConversion.from(new NullPointerException());
+                        }
+                        return SuccessfulConversion.fromSingle(o);
+                    } catch (Throwable t) {
+                        return FailedConversion.from(t);
                     }
-                    return SuccessfulConversion.fromSingle(o);
                 }
             });
         }

--- a/worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/platform/binding/ConsumeBindings.java
+++ b/worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/platform/binding/ConsumeBindings.java
@@ -27,6 +27,7 @@ import com.sk89q.worldedit.world.block.BaseBlock;
 import com.sk89q.worldedit.world.block.BlockState;
 import com.sk89q.worldedit.world.block.BlockStateHolder;
 import com.sk89q.worldedit.world.block.BlockType;
+import org.enginehub.piston.exception.StopExecutionException;
 import org.enginehub.piston.inject.InjectedValueAccess;
 
 import java.util.UUID;
@@ -49,36 +50,76 @@ public class ConsumeBindings extends Bindings {
     @Binding
     @Confirm(Confirm.Processor.REGION)
     public int regionMultiple(Actor actor, InjectedValueAccess context, @Selection Region region, String argument) {
-        int times = (int) Expression.compile(argument).evaluate();
-        return Confirm.Processor.REGION.check(actor, context, times);
+        try {
+            int times = (int) Expression.compile(argument).evaluate();
+            return Confirm.Processor.REGION.check(actor, context, times);
+        } catch (Throwable t) {
+            if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
+                throw t;
+            } else {
+                throw new InputParseException(t.getMessage(), t);
+            }
+        }
     }
 
     @Binding
     @Confirm(Confirm.Processor.RADIUS)
     public Integer radiusInteger(Actor actor, InjectedValueAccess context, String argument) {
-        int times = (int) Expression.compile(argument).evaluate();
-        return Confirm.Processor.RADIUS.check(actor, context, times);
+        try {
+            int times = (int) Expression.compile(argument).evaluate();
+            return Confirm.Processor.RADIUS.check(actor, context, times);
+        } catch (Throwable t) {
+            if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
+                throw t;
+            } else {
+                throw new InputParseException(t.getMessage(), t);
+            }
+        }
     }
 
     @Binding
     @Confirm(Confirm.Processor.LIMIT)
     public Integer limitInteger(Actor actor, InjectedValueAccess context, String argument) {
-        int times = (int) Expression.compile(argument).evaluate();
-        return Confirm.Processor.LIMIT.check(actor, context, times);
+        try {
+            int times = (int) Expression.compile(argument).evaluate();
+            return Confirm.Processor.LIMIT.check(actor, context, times);
+        } catch (Throwable t) {
+            if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
+                throw t;
+            } else {
+                throw new InputParseException(t.getMessage(), t);
+            }
+        }
     }
 
     @Binding
     @Confirm(Confirm.Processor.RADIUS)
     public Double radiusDouble(Actor actor, InjectedValueAccess context, String argument) {
-        double times = Expression.compile(argument).evaluate();
-        return Confirm.Processor.RADIUS.check(actor, context, times);
+        try {
+            double times = Expression.compile(argument).evaluate();
+            return Confirm.Processor.RADIUS.check(actor, context, times);
+        } catch (Throwable t) {
+            if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
+                throw t;
+            } else {
+                throw new InputParseException(t.getMessage(), t);
+            }
+        }
     }
 
     @Binding
     @Confirm(Confirm.Processor.LIMIT)
     public Double limitDouble(Actor actor, InjectedValueAccess context, String argument) {
-        double times = Expression.compile(argument).evaluate();
-        return Confirm.Processor.LIMIT.check(actor, context, times);
+        try {
+            double times = Expression.compile(argument).evaluate();
+            return Confirm.Processor.LIMIT.check(actor, context, times);
+        } catch (Throwable t) {
+            if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
+                throw t;
+            } else {
+                throw new InputParseException(t.getMessage(), t);
+            }
+        }
     }
 
     @Binding

--- a/worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/platform/binding/ConsumeBindings.java
+++ b/worldedit-core/src/main/java/com/fastasyncworldedit/core/extension/platform/binding/ConsumeBindings.java
@@ -57,7 +57,7 @@ public class ConsumeBindings extends Bindings {
             if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
                 throw t;
             } else {
-                throw new InputParseException(t.getMessage(), t);
+                throw new InputParseException(TextComponent.of(t.getMessage()), t);
             }
         }
     }
@@ -72,7 +72,7 @@ public class ConsumeBindings extends Bindings {
             if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
                 throw t;
             } else {
-                throw new InputParseException(t.getMessage(), t);
+                throw new InputParseException(TextComponent.of(t.getMessage()), t);
             }
         }
     }
@@ -87,7 +87,7 @@ public class ConsumeBindings extends Bindings {
             if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
                 throw t;
             } else {
-                throw new InputParseException(t.getMessage(), t);
+                throw new InputParseException(TextComponent.of(t.getMessage()), t);
             }
         }
     }
@@ -102,7 +102,7 @@ public class ConsumeBindings extends Bindings {
             if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
                 throw t;
             } else {
-                throw new InputParseException(t.getMessage(), t);
+                throw new InputParseException(TextComponent.of(t.getMessage()), t);
             }
         }
     }
@@ -117,7 +117,7 @@ public class ConsumeBindings extends Bindings {
             if (t instanceof StopExecutionException) { // Maintain throw from Confirm#check
                 throw t;
             } else {
-                throw new InputParseException(t.getMessage(), t);
+                throw new InputParseException(TextComponent.of(t.getMessage()), t);
             }
         }
     }

